File: job-portal-ios/JobPortal/Features/Jobs/Domain/Entities/Job.swift
import Foundation

struct Job: Identifiable {
    let id: String
    let title: String
    let company: String
    let location: String
    let workMode: WorkMode
    let employmentType: EmploymentType
    let experience: String
    let salary: Int?
    let description: String
    let requirements: [String]
    let skills: [String]
    let perks: [String]
    let employerId: String?
    let postedAt: Date

    enum WorkMode: String {
        case remote = "remote"
        case hybrid = "hybrid"
        case onsite = "onsite"
    }

    enum EmploymentType: String {
        case fulltime    = "fulltime"
        case parttime    = "parttime"
        case contract    = "contract"
        case internship  = "internship"
        case freelance   = "freelance"
    }
}

struct JobFilters {
    var workMode: Job.WorkMode?
    var employmentType: Job.EmploymentType?
    var location: String?
    var minimumSalary: Int?
}

File: job-portal-android/app/src/main/java/com/jobportal/features/jobs/domain/model/Job.kt
package com.jobportal.features.jobs.domain.model

data class Job(
    val id: String,
    val title: String,
    val company: String,
    val location: String,
    val description: String,
    val salary: Int?,
    val experience: String,
    val workMode: WorkMode,
    val employmentType: EmploymentType,
    val requirements: List<String>,
    val skills: List<String>,
    val perks: List<String>,
    val employerId: String,
    val postedAt: Long
)

enum class WorkMode(val value: String) {
    REMOTE("remote"), HYBRID("hybrid"), ONSITE("onsite");
    companion object {
        fun from(s: String?): WorkMode = entries.firstOrNull { it.value == s?.lowercase() } ?: ONSITE
    }
}

enum class EmploymentType(val value: String) {
    FULLTIME("fulltime"), PARTTIME("parttime"), CONTRACT("contract"),
    INTERNSHIP("internship"), FREELANCE("freelance");
    companion object {
        fun from(s: String?): EmploymentType = entries.firstOrNull { it.value == s?.lowercase() } ?: FULLTIME
    }
}
